### DISCUSSION TITLE
What happens to embeddings once i delete a file?

### DISCUSSION THREAD

**Discussion by abhuva** [brianpetro/obsidian-smart-connections#110] [Q&A]
I just wanted to ask what exactly happens to the embeddings if i delete (or modify) a file?
My guess is that its beeing updated (as i see constant calls to OpenAI Embeddings when i work on my vault).

For the future i think it would be great to also have a bit more control over this (if one wants) - by giving me the option to only update the embeddings when i want it (by using a command) - instead of doing it continuisly in the background.
It might save a bit on the tokens (altough embeddings seems to be quite cheap), but its also sometimes unneccessary if i just work on my vault, wich i do often (work on a file and delete it later once i am done).

**Comment by brianpetro** [comment]
Hi @abhuva 

One of my initial focuses was on making sure that notes weren't unnecessarily embedded, as I didn't want excess costs to prevent people from having the opportunity to utilize the technology.

This https://github.com/brianpetro/obsidian-smart-connections/issues/47#issuecomment-1454763138 explains some of the details.

Toggling on `log_render` in the advanced settings will show you token usage and which files are being embedded in the console log.

A file created and deleted before being closed and opened again should only use a negligible amount of tokens. You can double-check this with those console logs. 

Additionally, some of this depends on the underlying "embedding strategy," which I would consider still under active development. That "embedding strategy" plays a critical role in the performance of this plugin. This part also affects how the "efficiency layer" would need to be designed. But, as that "embedding strategy" gets refined, adding additional layers focused on efficiency will make more sense.

Something I have been considering implementing would be a "snooze" button, preventing new embeddings for a preset amount of time. Is that something you think would make sense for your use case?

Thanks,
Brian🌴

**Reply by abhuva** [reply]
thanks for taking the time and explain it.
I guess for my normal use-case i could also apply another strategy - that is excluding my inbox folder (wich would make sense anyway) as there are new files created. If i just use a file as some kind of workspace to delete later, it will never leave this inbox folder and therefor never get embedded anyway.

**Reply by brianpetro** [reply]
Great thinking! Thanks for sharing that strategy. 

Hopefully, someone else with a similar question will be able to make use of the exclusion feature in that way.

Brian🌴
